libs/parallelism/execution/include/hpx/execution/algorithms/when_all.hpp

```diff
@@ -244,20 +245,19 @@ namespace hpx { namespace execution { namespace experimental {
                 return operation_state<R, num_predecessors - 1>(
                     std::forward<R>(r), senders);
             }
-
-            template <typename R>
-            auto connect(R&& r) &
-            {
-                return operation_state<R, num_predecessors - 1>(r, senders);
-            }
```

Comment:
Because the constructor of the `operation_state` moves from the `senders`. The multi-sender connect is not supposed to do that.

Reply:
But they are copied here? The `connect(R&& r) &&` on the other hand does look buggy and should move `senders` into the operation state.

Reply:
Hmmm, I might miss something. I thought that the elements of the senders tuple are moved one-by-one here: 
https://github.com/STEllAR-GROUP/hpx/blob/master/libs/parallelism/execution/include/hpx/execution/algorithms/when_all.hpp#L156
and here: 
https://github.com/STEllAR-GROUP/hpx/blob/master/libs/parallelism/execution/include/hpx/execution/algorithms/when_all.hpp#L224

Reply:
Good point, you're right. So in *that* case, the `senders` parameter in the constructors should just be moved or forwarded depending on which `connect` is called. I can take care of this separately, or if you'd like you're of course free to make the change here.

Reply:
> might be good to separate the changes anyway since they're not related to `bulk`.

Yep, This really was a flyby change. I tried fixing it properly by doing what you suggested, but I couldn't get it quite right (I haven't spent a ton of time on refactoring things, however). Let's create a ticket and solve it separately.

Reply:
Thanks for spotting it though! I'll create an issue.

Reply:
https://github.com/STEllAR-GROUP/hpx/issues/5423